generator/src/test/java/com/scottlogic/deg/generator/restrictions/NumericRestrictionsMergerTests.java

```diff
@@ -0,0 +1,105 @@
+
+    @Test
+    public void merge_withOnlyRightNumericRestrictions_shouldReturnLeftRestrictions(){
+        NumericRestrictionsMerger merger = new NumericRestrictionsMerger();
+        NumericRestrictions right = new NumericRestrictions();
+
+        MergeResult<NumericRestrictions> result = merger.merge(null, right);
+
+        Assert.assertThat(result, not(nullValue()));
+        Assert.assertThat(result.successful, is(true));
+        Assert.assertThat(result.restrictions, is(sameInstance(right)));
+    }
+
+    @Test
+    public void merge_withNonContradictoryNumericRestrictions_shouldReturnMergedRestrictions(){
+        NumericRestrictionsMerger merger = new NumericRestrictionsMerger();
+        NumericRestrictions left = new NumericRestrictions();
+        NumericRestrictions right = new NumericRestrictions();
+        left.min = new NumericLimit<>(BigDecimal.ZERO, true);
+        left.max = new NumericLimit<>(BigDecimal.TEN, true);
+        right.min = new NumericLimit<>(BigDecimal.ONE, false);
+        right.max = new NumericLimit<>(BigDecimal.TEN, false);
+
+        MergeResult<NumericRestrictions> result = merger.merge(left, right);
+
+        Assert.assertThat(result, not(nullValue()));
+        Assert.assertThat(result.successful, is(true));
+        Assert.assertThat(result.restrictions, not(nullValue()));
+        Assert.assertThat(result.restrictions.min.getLimit(), is(equalTo(BigDecimal.ONE)));
+        Assert.assertThat(result.restrictions.min.isInclusive(), is(false));
+        Assert.assertThat(result.restrictions.max.getLimit(), is(equalTo(BigDecimal.TEN)));
+        Assert.assertThat(result.restrictions.max.isInclusive(), is(false));
+    }
+
+    @Test
+    public void merge_withLessThanOrEqualAndGreaterThanOrEqualSameNumber_shouldReturnMergedRestrictions(){
+        NumericRestrictionsMerger merger = new NumericRestrictionsMerger();
+        NumericRestrictions greaterThanOrEqual = new NumericRestrictions();
+        NumericRestrictions lessThanOrEqual = new NumericRestrictions();
+        greaterThanOrEqual.min = new NumericLimit<>(BigDecimal.TEN, true);
+        lessThanOrEqual.max = new NumericLimit<>(BigDecimal.TEN, true);
+
+        MergeResult<NumericRestrictions> result = merger.merge(greaterThanOrEqual, lessThanOrEqual);
+
+        Assert.assertThat(result, not(nullValue()));
+        Assert.assertThat(result.successful, is(true));
+        Assert.assertThat(result.restrictions, not(nullValue()));
+        Assert.assertThat(result.restrictions.min.getLimit(), is(equalTo(BigDecimal.TEN)));
+        Assert.assertThat(result.restrictions.min.isInclusive(), is(true));
+        Assert.assertThat(result.restrictions.max.getLimit(), is(equalTo(BigDecimal.TEN)));
+        Assert.assertThat(result.restrictions.max.isInclusive(), is(true));
+    }
+
+    @Test
+    public void merge_withContradictoryNumericRestrictions_shouldReturnUnsuccessful(){
+        NumericRestrictionsMerger merger = new NumericRestrictionsMerger();
```

Comment:
Similar to above, seems like `merger` is always needed in each test, is it better for it to be in a `beforeEach`?